R/zchunk_LB162.ag_prodchange_R_C_Y_GLU_irr.R

```diff
@@ -41,67 +45,460 @@ module_aglu_LB162.ag_prodchange_R_C_Y_GLU_irr_DISABLED <- function(command, ...)
+      mutate(Irr_Rfd = "RFD") %>%
+      rename(yield_kgHa = Yield_kgHa_rainfed) %>%
+      bind_rows(L162.ag_irrYield_kgHa_Rcrs_Ccrs_Y ) %>%
+      # add the missing SPEC_AG_PROD_YEARS and interpolate the yields
+      tidyr::complete(year = c(year, SPEC_AG_PROD_YEARS) ,
+                      CROSIT_ctry, CROSIT_crop, Irr_Rfd) %>%
+      select(CROSIT_ctry, CROSIT_crop, Irr_Rfd, year, yield_kgHa) %>%
+      arrange(year) %>%
+      group_by(CROSIT_ctry, CROSIT_crop, Irr_Rfd) %>%
+      mutate(yield_kgHa = approx_fun(year, yield_kgHa)) %>%
+      ungroup() %>% filter(year %in% SPEC_AG_PROD_YEARS) %>%
+      # calculate yield multipliers
+      group_by(CROSIT_ctry, CROSIT_crop, Irr_Rfd) %>%
+      mutate(Mult = yield_kgHa / first(yield_kgHa)) %>%
+      ungroup() %>%
+      select(-yield_kgHa) %>%
+      # Drop the NaN's = crops with zero base year production / harvested area
+      na.omit() ->
+      L162.ag_Yieldmult_Rcrs_Ccrs_Y_irr
+
+
+    # We apply the above yield multipliers to crop-specific changes in harvested area.
+    # This removes bias from changes in composition of GCAM commodities in the FAO projections.
+    # These yield multipliers are now ready to be matched into the GTAP/LDS-based table of
+    # country x crop x zone harvested area in the base year, L151.ag_irrHA/rfdHA...
     #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
+    # First, merge the separate rainfed and irrigated harvested area datasets to simplify processing.
+    # Then join CROSIT country and crop identifiers. This intermediate table is used in multiple
+    # subsequent pipelines.
     #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
+    # CROSIT_ctry identifiers come from the AGLU_ctry table, which needs preprocessing to avoid unwanted
+    # extra rows due to Yugoslav FSR.
+
+    # deal with unfilled Yugoslavia entries in AGLU_ctry without impacting other code chunks.
+    AGLU_ctry %>%
+      select(iso, CROSIT_ctry) %>%
+      filter(!is.na(CROSIT_ctry)) %>%
+      dplyr::distinct() ->
+      AGLU_ctry_iso_CROSIT
+
+    L151.ag_irrHA_ha_ctry_crop %>%
+      mutate(Irr_Rfd = "IRR") %>%
+      rename(HA = irrHA) ->
+      L151.ag_irrHA_ha_ctry_crop
+    L151.ag_rfdHA_ha_ctry_crop %>%
+      mutate(Irr_Rfd = "RFD") %>%
+      rename(HA = rfdHA) %>%
+      bind_rows(L151.ag_irrHA_ha_ctry_crop) %>%
+      # Join CROSIT country and crop information and aggregate
+      # keeping NA's for later processing
+      left_join(AGLU_ctry_iso_CROSIT, by = "iso") %>%
+      left_join(dplyr::distinct(select(FAO_ag_items_PRODSTAT, CROSIT_crop, GTAP_crop)),
+                               by = "GTAP_crop") ->
+      L162.ag_HA_ha_ctry_crop_irr # agrees to 10e-8
```

Comment:
Could you expand on the comment? I don't know what `agrees to 10e-8` means. I'm guessing you are saying this table would fail the oldnew test, but does actually match. If so, say that explicitly.